src/main/scala/it/pagopa/interop/authorizationprocess/service/impl/AuthorizationManagementServiceImpl.scala

```diff
@@ -68,7 +77,11 @@ final case class AuthorizationManagementServiceImpl(
         clientApi.deleteClient(xCorrelationId = correlationId, clientId.toString, xForwardedFor = ip)(
           BearerToken(bearerToken)
         )
-      invoker.invoke(request, "Client delete")
+      invoker
+        .invoke(request, "Client delete")
+        .recoverWith {
+          case err: ApiError[_] if err.code == 404 => Future.failed(ClientNotFound(clientId))
```

Comment:
How about a partial function with this logic? Since you're abstracting everything it will make sense. It can even be a syntax addition over the type of `invoker.invoke(request, "Client delete")` like `.withNotFound(ClientNotFound(clientId))`

Reply:
probably I didn't get your comment correctly, because it already is a partial function. Could you give more details?

The syntax addition (I guess you mean like an implicit class) could be a nice syntax sugar, but the `ApiError` is different for each service, and we should create one class per service.
Aren't we risking to create even more boilerplate?

Or you meant a different solution?

Reply:
No you got that, while I didn't realize that each service has is own ApiError. One refactor that might be investigated is the chance to move ApiError itself in commons to be able to generalize these things maybe.